src/reconcile/rec_visibility.c

```diff
@@ -292,6 +292,36 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
         if (WT_TXNID_LT(max_txn, txnid))
             max_txn = txnid;
 
+        /*
```

Comment:
There is another problem in obsolete check. We will not discard the U@10 in the case U@5 -> U@10 if U@5 is globally visible but U@10 is not. I'm not sure how this will impact this change but it seems the assumption is wrong.

If the previous update is globally visible, we don't need to care about out-of-order ids or timestamps since the offending update will be removed by an obsolete check further down the line.

Reply:
Hmm good catch, I didn't realise the check worked like that. That's quite significant.

The problem is that I need some way of figuring out when I can unpin the updates. It seems that I may have to actually check that everything behind it is globally visible in order to ensure that.

Reply:
As mentioned below, I think this is fine.
I've amended the comment to describe this more accurately.